plugins/base/src/main/kotlin/renderers/html/HtmlRenderer.kt

```diff
@@ -764,138 +763,35 @@ open class HtmlRenderer(
-                        else templateCommandAsHtmlComment(PathToRootSubstitutionCommand(TEMPLATE_REPLACEMENT, default = pathToRoot)) {
-                            link(href = TEMPLATE_REPLACEMENT + it)
-                        }
-                        else -> unsafe { +it }
-                    }
-                }
-            }
-            body {
-                div("navigation-wrapper") {
-                    id = "navigation-wrapper"
-                    div {
-                        id = "leftToggler"
-                        span("icon-toggler")
-                    }
-                    div("library-name") {
-                        clickableLogo(page, pathToRoot)
-                    }
-                    div { templateCommand(ReplaceVersionsCommand(path.orEmpty())) }
-                    div("pull-right d-flex") {
-                        filterButtons(page)
-                        button {
-                            id = "theme-toggle-button"
-                            span {
-                                id = "theme-toggle"
-                            }
-                        }
-                        div {
-                            id = "searchBar"
-                        }
-                    }
-                }
-                div {
-                    id = "container"
-                    div {
-                        id = "leftColumn"
-                        div {
-                            id = "sideMenu"
-                        }
-                    }
-                    div {
-                        id = "main"
-                        content()
-                        div(classes = "footer") {
-                            span("go-to-top-icon") {
-                                a(href = "#content") {
-                                    id = "go-to-top-link"
-                                }
-                            }
-                            span {
-                                configuration?.footerMessage?.takeIf { it.isNotEmpty() }
-                                    ?.let { unsafe { raw(it) } }
-                                    ?: text(defaultFooterMessage)
-                            }
-                            span("pull-right") {
-                                span { text("Generated by ") }
-                                a(href = "https://github.com/Kotlin/dokka") {
-                                    span { text("dokka") }
-                                    span(classes = "padded-icon")
-                                }
-                            }
-                        }
+
+    private val templateModelFactory = TemplateModelFactory(context)
+    private val templater = HtmlTemplater(context).apply { setupSharedModel(templateModelFactory.buildSharedModel()) }
+
+    open fun buildHtml(page: PageNode, resources: List<String>, content: FlowContent.() -> Unit): String =
+        templater.renderFromTemplate(DokkaTemplateTypes.BASE) {
```

Comment:
I was under the impression that plugin makers will be able to overwrite `base.ftl` and have fully customizable html pages, but seeing hardcoded `renderFromTemplate(DokkaTemplateTypes.BASE)` and `ClassTemplateLoader(javaClass, "/dokka/templates")` in another place makes me think that I was wrong

If I make a plugin, copy `base.ftl` with some minor changes (remove footer completely), put it in a folder and specify it in `templatesDir` - which template and thus frontend will be used? Is it somehow both of them?

Reply:
Oh, I see, so the user's template has to be called `base.ftl` as well. Can we stress it in documentation somewhere? I thought it could have any name you want